### PR TITLE
[WIP] Aggregation tsDao test

### PR DESCRIPTION
**Pull request by van-vanich**
When we use old realization for case(startTs = 0, endTs = 3000, interval = 2000), we have issue: 
image
We can see program return results for periods 0-2000 and 2000 - 4000 but need 0-2000 and 2000-3000.

Also, if we see the realization SQL request, for period startTs-endTs, we return result in [startTs; endTs). And we don't return data for endTs. In case(startTs = 0, endTs = 4000, interval = 2000) we return data for [0-2000) and [2000-4000) in old realization. So we don't return data which have Ts = 4000.

### CODE REVIEW

File: dao/src/test/java/org/thingsboard/server/dao/service/timeseries/BaseTimeseriesServiceTest.java [hunk 1]
```diff
@@ -164,32 +164,138 @@ public void testFindByQueryAscOrder() throws Exception {
...
-//        Assert.assertEquals(3, entries.size());
-//        Assert.assertEquals(toTsEntry(TS - 3, stringKvEntry), entries.get(0));
-//        Assert.assertEquals(toTsEntry(TS - 2, stringKvEntry), entries.get(1));
-//        Assert.assertEquals(toTsEntry(TS - 1, stringKvEntry), entries.get(2));
-//    }
+
+    @Test
+    public void testFindByQueryOneMilisecondPeriod() throws Exception {
+        DeviceId deviceId = new DeviceId(Uuids.timeBased());
+        saveEntries(deviceId, TS);
```
Comment (smatvienko-tb) [inline dao/src/test/java/org/thingsboard/server/dao/service/timeseries/BaseTimeseriesServiceTest.java:171]: Please, cover the corner cases. Put additional data point before and after test period.
```
saveEntries(deviceId, TS-1);
saveEntries(deviceId, TS);
saveEntries(deviceId, TS+1);
```

File: dao/src/test/java/org/thingsboard/server/dao/service/timeseries/BaseTimeseriesServiceTest.java [hunk 2]
```diff
@@ -164,32 +164,138 @@ public void testFindByQueryAscOrder() throws Exception {
...
+
+        entries = tsService.findAll(tenantId, entityView.getId(), queries).get();
+        Assert.assertEquals(1, entries.size());
+        Assert.assertEquals(toTsEntry(TS, new LongDataEntry(LONG_KEY, 1L)), entries.get(0));
+    }
+
+    @Test
+    public void testFindByQuery_whenPeriodEqualsInterval() throws Exception {
+        DeviceId deviceId = new DeviceId(Uuids.timeBased());
+        for (long i=TS; i<= TS + 10L; i++) {
```
Comment (smatvienko-tb) [inline dao/src/test/java/org/thingsboard/server/dao/service/timeseries/BaseTimeseriesServiceTest.java:190]: Fill additional datapoints on startTs-1 and endTs+1 to cover corner cases

File: dao/src/test/java/org/thingsboard/server/dao/service/timeseries/BaseTimeseriesServiceTest.java [hunk 3]
```diff
@@ -164,32 +164,138 @@ public void testFindByQueryAscOrder() throws Exception {
...
+    }
+
+    @Test
+    public void testFindByQuery_whenPeriodHaveTwoInterval_whereSecondShorterThanFirst() throws Exception {
+        DeviceId deviceId = new DeviceId(Uuids.timeBased());
+        for (long i=TS; i<= TS + 8L; i++) {
+            saveEntries(deviceId, i);
+        }
+
+        List<ReadTsKvQuery> queries = new ArrayList<>();
```
Comment (smatvienko-tb) [inline dao/src/test/java/org/thingsboard/server/dao/service/timeseries/BaseTimeseriesServiceTest.java:238]: List.of()

File: dao/src/test/java/org/thingsboard/server/dao/service/timeseries/BaseTimeseriesServiceTest.java [hunk 4]
```diff
@@ -164,32 +164,138 @@ public void testFindByQueryAscOrder() throws Exception {
...
+        queries.add(new BaseReadTsKvQuery(LONG_KEY, TS, TS + 7, 5, 1, Aggregation.COUNT, DESC_ORDER));
```
Comment (smatvienko-tb) [inline dao/src/test/java/org/thingsboard/server/dao/service/timeseries/BaseTimeseriesServiceTest.java:239]: the numbers 8,7,5,2,6,3 are confusing. Please, use the numbers that are easy to understand for humans. Like from 10 to 20, 100 to 300. Then I can easy to understand whether you do +1 and -1 for tests

File: dao/src/test/java/org/thingsboard/server/dao/service/timeseries/BaseTimeseriesServiceTest.java [hunk 5]
```diff
@@ -164,32 +164,138 @@ public void testFindByQueryAscOrder() throws Exception {
...
-//
-//        entries = tsService.findAll(tenantId, entityView.getId(), queries).get();
-//        Assert.assertEquals(3, entries.size());
-//        Assert.assertEquals(toTsEntry(TS - 3, stringKvEntry), entries.get(0));
-//        Assert.assertEquals(toTsEntry(TS - 2, stringKvEntry), entries.get(1));
-//        Assert.assertEquals(toTsEntry(TS - 1, stringKvEntry), entries.get(2));
-//    }
+
+    @Test
+    public void testFindByQueryOneMilisecondPeriod() throws Exception {
```
Comment (smatvienko-tb) [inline dao/src/test/java/org/thingsboard/server/dao/service/timeseries/BaseTimeseriesServiceTest.java:169]: Please, do the same naming convention for all new methods

**Review by smatvienko-tb — COMMENTED** [review]
Please, consider to make changes regarding this review

**Review by smatvienko-tb — APPROVED** [review]
This ts code was refactored as required. and new tests were developed to test corner cases.
Good job.
But please, merge your branch with the latest master and resolve conflicts before.
